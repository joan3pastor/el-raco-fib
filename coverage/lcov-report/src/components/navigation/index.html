<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/navigation</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> src/components/navigation
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.01% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>337/432</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.43% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>191/221</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.61% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>77/141</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75.74% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>281/371</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Assignatures.js"><a href="Assignatures.js.html">Assignatures.js</a></td>
	<td data-value="82.35" class="pic high"><div class="chart"><div class="cover-fill" style="width: 82%;"></div><div class="cover-empty" style="width:18%;"></div></div></td>
	<td data-value="82.35" class="pct high">82.35%</td>
	<td data-value="34" class="abs high">28/34</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="22" class="abs high">20/22</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="16" class="abs medium">12/16</td>
	<td data-value="82.35" class="pct high">82.35%</td>
	<td data-value="34" class="abs high">28/34</td>
	</tr>

<tr>
	<td class="file high" data-value="Avisos.js"><a href="Avisos.js.html">Avisos.js</a></td>
	<td data-value="81.98" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.98" class="pct high">81.98%</td>
	<td data-value="172" class="abs high">141/172</td>
	<td data-value="91.45" class="pct high">91.45%</td>
	<td data-value="117" class="abs high">107/117</td>
	<td data-value="54.29" class="pct medium">54.29%</td>
	<td data-value="35" class="abs medium">19/35</td>
	<td data-value="77.17" class="pct medium">77.17%</td>
	<td data-value="127" class="abs medium">98/127</td>
	</tr>

<tr>
	<td class="file low" data-value="Drawer.js"><a href="Drawer.js.html">Drawer.js</a></td>
	<td data-value="47.37" class="pic low"><div class="chart"><div class="cover-fill" style="width: 47%;"></div><div class="cover-empty" style="width:53%;"></div></div></td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="47.37" class="pct low">47.37%</td>
	<td data-value="19" class="abs low">9/19</td>
	</tr>

<tr>
	<td class="file high" data-value="Events.js"><a href="Events.js.html">Events.js</a></td>
	<td data-value="81.03" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.03" class="pct high">81.03%</td>
	<td data-value="58" class="abs high">47/58</td>
	<td data-value="67.65" class="pct medium">67.65%</td>
	<td data-value="34" class="abs medium">23/34</td>
	<td data-value="86.67" class="pct high">86.67%</td>
	<td data-value="15" class="abs high">13/15</td>
	<td data-value="78.43" class="pct medium">78.43%</td>
	<td data-value="51" class="abs medium">40/51</td>
	</tr>

<tr>
	<td class="file low" data-value="ExternalLinks.js"><a href="ExternalLinks.js.html">ExternalLinks.js</a></td>
	<td data-value="20" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="15" class="abs low">3/15</td>
	</tr>

<tr>
	<td class="file low" data-value="Feedback.js"><a href="Feedback.js.html">Feedback.js</a></td>
	<td data-value="40" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	</tr>

<tr>
	<td class="file high" data-value="Horari.js"><a href="Horari.js.html">Horari.js</a></td>
	<td data-value="89.86" class="pic high"><div class="chart"><div class="cover-fill" style="width: 89%;"></div><div class="cover-empty" style="width:11%;"></div></div></td>
	<td data-value="89.86" class="pct high">89.86%</td>
	<td data-value="69" class="abs high">62/69</td>
	<td data-value="86.11" class="pct high">86.11%</td>
	<td data-value="36" class="abs high">31/36</td>
	<td data-value="69.23" class="pct medium">69.23%</td>
	<td data-value="13" class="abs medium">9/13</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="66" class="abs high">60/66</td>
	</tr>

<tr>
	<td class="file high" data-value="Labs.js"><a href="Labs.js.html">Labs.js</a></td>
	<td data-value="81.25" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.25" class="pct high">81.25%</td>
	<td data-value="16" class="abs high">13/16</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="Noticies.js"><a href="Noticies.js.html">Noticies.js</a></td>
	<td data-value="71.43" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="42" class="abs medium">30/42</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="45" class="pct low">45%</td>
	<td data-value="20" class="abs low">9/20</td>
	<td data-value="71.05" class="pct medium">71.05%</td>
	<td data-value="38" class="abs medium">27/38</td>
	</tr>

<tr>
	<td class="file high" data-value="SalesBibl.js"><a href="SalesBibl.js.html">SalesBibl.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jul 17 2020 06:12:00 GMT+0200 (GMT+02:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
<script src="..\..\..\block-navigation.js"></script>
</body>
</html>
