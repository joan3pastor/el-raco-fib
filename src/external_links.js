// OAUTH:
export const authURL = 'https://api.fib.upc.edu/v2/o/authorize';
export const tokenURL = 'https://api.fib.upc.edu/v2/o/token';
export const clientID = '8IVfHurpzfHhC68LdfpIoRoh7uWcr5bXBkCGroV4';
export const clientSECRET = '8W3Xbf3kRfIQK3rW8QR2xED3TxZ1cI3kMv4VnqyeoreruGNKXBnL8MtcJUztndI7QiuqvJhkvfNE53fOSfbqOT4DEH75d2TmgrmPTHeU5HDuRrA2tFlyYSBoKH31O9K4';
export const callbackURI = 'apifib://callback'
export const revokeURL = "https://api.fib.upc.edu/v2/o/revoke_token/";

// API URLs:
export const joURL = 'https://api.fib.upc.edu/v2/jo/';
export const fotoURL = 'https://api.fib.upc.edu/v2/jo/foto.jpg';
export const avisosURL = 'https://api.fib.upc.edu/v2/jo/avisos/';
export const horariURL = 'https://api.fib.upc.edu/v2/jo/classes/';
export const assigURL = 'https://api.fib.upc.edu/v2/jo/assignatures/';
export const noticiesURL = 'https://api.fib.upc.edu/v2/noticies/';
export const imglabsURL = 'https://api.fib.upc.edu/v2/laboratoris/imatges/';
export const eventsURL = 'https://api.fib.upc.edu/v2/events/';
export const practiquesURL = 'https://api.fib.upc.edu/v2/jo/practiques/';
export const placesURL = 'https://api.fib.upc.edu/v2/assignatures/places/';
export const mobilitatURL = 'https://api.fib.upc.edu/v2/universitats-erasmus/';
export const lecturesURL = 'https://api.fib.upc.edu/v2/lectures/properes/';
export const QActualURL = 'https://api.fib.upc.edu/v2/quadrimestres/actual/';
export const aulesURL = 'https://api.fib.upc.edu/v2/aules/';

// UPC Notícies RSS Channel
export const upcNoticiesRSSca = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";
export const upcNoticiesRSSes = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";
export const upcNoticiesRSSen = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";

// Reservas Bibl. 
export const salesBiblioteca = 'https://apps.bibliotecnica.upc.edu/reserva_sales/comunes/disponibilidad.php?Id_centro=1&Admin=1&Parametros=TnVtX2NvbHVtbmFzPTYmQWx0bz0mQW5jaG89MTAwJSZGdWVudGU9QXJpYWwmVGFtYW5vX2Z1ZW50ZT0xOHB4JlRhbWFub19jb250b3Jubz0ycHgmTWFyZ2VuX2RvY3VtZW50bz0wcHgmQ29sb3JfZm9uZG89ZmZmZmZmJlBhZGRpbmdfY2VsZGE9MTBweCZNb3N0cmFyX3RpdHVsbz0xJkFicmV2aWF0dXJhcz0xJk1vc3RyYXJfbGlua19hZG1pbj0wJk1vc3RyYXJfdGV4dG9fZGlzcG9uaWJsZXM9MCZDb21wYWN0YXJfdWJpY2FjaW9uZXM9MCZNb3N0cmFyX2VubGFjZV9pbmZvcm1hY2lvbj0wJk1vc3RyYXJfYm90b25fcmVzZXJ2YXI9MQ==';

// Enllaços d'interès
export const linkDEFIB = 'http://defib.upc.edu/qui_som';
export const linkAssociacions = 'https://www.fib.upc.edu/ca/la-fib/vida-universitaria/associacions';
export const linkEduroam = 'https://serveistic.upc.edu/ca/wifiupc/documentacio/assistent-de-configuracio-per-eduroam';
export const linkCorreu = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/correu-destudiant';
export const linkImpressio = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/servei-dimpressio';
export const linkUPClink = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/vpn-upclink';
export const linkAcordsUPC = 'https://serveistic.upc.edu/ca/distsoft/el-servei/programari-per-a-estudiantsi-professors';
export const linkVMware = 'https://raco.fib.upc.edu/vmware';
export const linkGithub = 'https://education.github.com/pack';
export const linkAzure = 'https://azure.microsoft.com/en-us/free/students/';
export const linkOffice = 'https://www.microsoft.com/es-es/education/products/office';
export const linkGCP = 'https://cloud.google.com/edu/';

// Feedback
export const playstorelink = 'https://play.google.com/store/apps/details?id=com.fibupc';

// Programes de Mobilitat
export const mobInfoGeneral = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/estades-academiques';
export const mobSessionsInfo = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/sessions-informatives';
export const mobCalendari = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/calendari-mobilitat';
export const mobBeques = 'https://www.upc.edu/sga/ca/Beques/BequesMobilitat';

// Github
export const githubURL = "https://github.com/joan3pastor/el-raco-fib";
