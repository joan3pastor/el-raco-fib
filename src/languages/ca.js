export const ca = {
    code: "ca",
    bottomTabs: { 
        horari: "HORARI",
        avisos: "AVISOS",
        noticies: "NOTÍCIES",
        assig: "ASSIG.",
    },
    titleViews: { 
        events: "Esdeveniments", 
        notes: "Gestor de Notes", 
        labs: "Disponibilitat Labs", 
        bibl: "Sales Biblioteca BRGF", 
        aules: "Cerca Aules Lliures", 
        places: "Seguiment Places Matr.", 
        mobilitat: "Programes de Mobilitat", 
        lectures: "Lectures TFG i TFM", 
        links: " Enllaços d'Interès ", 
        feedback: "Feedback & Bugs", 
        settings: "Configuració", 
    },
    drawer: { 
        events: "Esdeveniments",
        notes: "Gestor de Notes",
        labs: "Disponibilitat Labs",
        bibl: "Sales Bibl. BRGF",
        aules: "Cerca Aules Lliures",
        places: "Places Matrícula",
        mobilitat: "Programes de Mobilitat",
        lectures: "Lectures TFG i TFM",
        links: "Enllaços d'Interès",
        feedback: "Feedback & Bugs",
        settings: "Configuració",
        logout: "Log Out",
    },
    horari: { 
        grup: "Grup",
        aula: "Aula",
        diesSetmana: ["dl.", "dt.", "dc.", "dj.", "dv."],
    },
    avisos: {
        msgEmpty: "No tens avisos.",
    }, 
    noticies: {}, 
    assig: { 
        grup: "Grup",
        credits: "Credits",
        semestre: "Semestre",
        professors: "Professors",
        avaluacio: "Mètode d'avaluació",
        metodologia: "Metodologia docent",
        loadingProfessors: "Carregant professors...",
        loadingAvaluacio: "Carregant avaluació...",
        loadingMetodologia: "Carregant metodologia...",
        responsable: "Responsable",
        msgEmpty: "No estàs cursant cap assignatura.",
    },
    events: { 
        footer: "Només mostra exàmens en hores no lectives i dies festius fins a 3 mesos endevant.",
    },
    notes: { 
        mitja: "Nota mitja",
        final: "Nota total",
        progres: "Progrés",
        bttnAfegir: "AFEGIR ENTRADA",
        bttnCancellar: "CANCEL·LAR",
        bttnGuardar: "GUARDAR",
        aeNom: "Nom acte avaluatiu",
        aePes: "Pes (0 a 1)",
        aeNota: "Nota",
        textNota: "Nota",
        titolEliminar: "Atenció!",
        textEliminar: "Segur que vols eliminar l'entrada?  ",
        textSi: "SI",
        textNo: "NO",
        empty: "No estàs cursant cap assignatura.",
    },
    labs: { 
        bttnActualitzar: "ACTUALITZAR",
    },
    bibl: {}, 
    aules: { 
        pickerDiaNom: "Dia",
        pickerIniciNom: "Inici",
        pickerDuradaNom: "Durada",
        pickerTipusNom: "Tipus",
        pickerDia: ["Avui", "dl.", "dt.", "dc.", "dj.", "dv."],
        pickerDiaFormat: [{value:"Avui"}, {value:"dl."}, {value:"dt."}, {value:"dc."}, {value:"dj."}, {value:"dv."}],
        pickerTipus: ["Tots", "AULA", "LAB"],
        pickerTipusFormat: [{value:"Tots"}, {value:"AULA"}, {value:"LAB"}],
        msgSelecDia: "Selecciona un dia.",
        msgSelecInici: "Selecciona una hora d'inici.",
        msgSelecDurada: "Selecciona una durada mínima.",
        sectionAules: "Aules lliures",
        sectionLabs: "Labs lliures",
        aulaMsgLinia1: "Lliure fins",
        aulaMsgLinia2: "les",
    },
    places: { 
        bttnAfegir: "Afegir Assig.",
        bttnActualitzar: "Actualitzar",
        msgActualitzant: "Actualitzant en",
        msgBuscarAssig: "Buscar Assignatura",
        msgGrup: "Grup",
        cursos: {
            "GRAU": "Grau en Enginyeria Informàtica",
            "GCED": "Grau en Ciència i Enginyeria de Dades",
            "MEI": "Màster en Enginyeria Informàtica",
            "MIRI": "Màster en Innovació i Recerca en Informàtica",
            "MSEC": "Màster en Formació ESO i Batxillerat",
            "MAI": "Màster en Intel·ligència Artificial",
        },
        titolEliminar: "Atenció!",
        textEliminar: "Segur que vols eliminar l'entrada?  ",
        textSi: "SI",
        textNo: "NO",
    },
    mobilitat: { 
        pickerPais: "País",
        pickerPrograma: "Programa",
        pickerElemTots: "Tots",
        modalTitol: "Enllaços d'Interès",
        modalLink1: "Informació General",
        modalLink2: "Sessions Informatives",
        modalLink3: "Calendari",
        modalLink4: "Beques",
        bttnWeb: "Pàgina Web",
        bttnMapa: "Obrir Mapa",
        bttnPDF: "PDF Info",
        msgPais: "País",
        msgPrograma: "Programa",
        msgVigent: "Vigent fins",
    },
    lectures: { 
        msgBuit: "No hi ha lectures programades.",
        pickerPla: "Pla",
        pickerElemTots: "Tots",
        director: "Director",
        estudiant: "Estudiant",
        dia: "Dia",
        lloc: "Lloc",
        modalitat: "Modalitat",
        pla: "Pla",
    },
    links: { 
        title1: "La Facultat",
        title2: "Guies",
        title3: "Programari i Serveis per a l'Estudiant",
        A1E1: "DEFIB (Delegació d'Estudiants de la FIB)",
        A1E1Comentari: "   - Bona secció de FAQ per a nous estudiants.",
        A1E2: "Associacions",
        A2E1: "Manual de connexió a Eduroam (Wi-Fi)",
        A2E2: "Configuració del correu d'estudiant",
        A2E3: "Servei d'impressió",
        A2E4: "Configuració UPClink (VPN per accedir als servidors de la FIB des de fora la facultat)",
        A3E1: "Acords de distribució de programari UPC (Microsoft Imagine, Autocad, Matlab...)",
        A3E2: "VMware Academic Program (Ofert per la FIB)",
        A3E3: "Github Student Developer Pack (Github private projects, ASW Educate, Heroku Free Dynos, etc.)",
        A3E4: "Microsoft Azure for Students",
        A3E5: "Microsoft Office for Students + 1 TB OneDrive",
        A3E6: "Google Cloud Platform Education Credits",
        msgManteniment: "Si algun enllaç deixa de funcionar envieu-me un correu des de l'apartat \"Feedback & Bugs\".",
    },
    feedback: { 
        text: "<style>p{text-align:justify}</style><p>Siéntete libre de enviarme un correo si algo no se comporta como esperabas o si echas en falta alguna funcionalidad. Adjunta capturas de pantalla de todo lo posible.</p><p>Trataré de solucionar errores críticos. Todo lo demás dependerá de mi disponibilidad.</p>",
        bttnSendMail: "Envíame un correo!",
        bttnValora: "Valora la App",
        bttnGithub: "Colabora!",
        footer: "Creada por Joan Marc Pastor con la API del Racó",
    },
    settings: { 
        A1Title: "Notificacions avisos",
        A1Bttn1: "Activades",
        A1Bttn2: "Desactivades",
        A1Nota1: "Es buscaran nous avisos cada 20 minuts.",
        A1Nota2: "*Pot no funcionar en \"Power Saving Mode\".",
        A2Title: "Mida de la font dels avisos",
        A2text: "Text de prova...",
        A3Title: "Estil Horari",
        A3Bttn1: "Complet",
        A3Bttn2: "Minimalista",
        A4Title: "Idioma",
        A4Placeholder: "Selecciona idioma",
        dies: ["dl.", "dt.", "dc.", "dj.", "dv."],
        pickerDies: [{value:"dl."}, {value:"dt."}, {value:"dc."}, {value:"dj."}, {value:"dv."}],
        A5Title: "Modificar Horari",
        A5Bttn1: "Afegir Classe",
        A5Bttn2: "Eliminar Classe",
        A5Bttn3: "Reset",
        A5txt1: "Classes afegides",
        A5txt2: "Classes eliminades",
        A5ModalTitle: "Afegir classe",
        A5ModalP1: "Codi assig.",
        A5ModalP2: "Aula",
        A5ModalP3: "Dia",
        A5ModalP4: "Inici",
        A5ModalP5: "Durada",
        A5ModalP6: "Grup",
        A5ModalP7: "Tipus",
        A5ModalBttn1: "CANCEL·LAR",
        A5ModalBttn2: "GUARDAR",
        A5Modal2Title: "Eliminar classes",
        A5Modal2txt1: "Classes oficials",
        A5Modal2txt2: "Classes afegides",
        A5Alert1: "Introdueix el codi de l'assignatura.",
        A5Alert2: "Introduce la aula de la clase.",
        A5Alert3: "Introdueix la aula de la classe.",
        A5Alert4: "Introdueix l'hora d'inici de la classe.",
        A5Alert5: "Introdueix la durada de la classe.",
        A5Alert6: "Introdueix el grup.",
        A5Alert7: "Introdueix el tipus de la classe. \n\nT = Teoria\nL = Lab\nP = Problemes",
    },    
}
