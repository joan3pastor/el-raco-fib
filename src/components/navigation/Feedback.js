import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Text, View, ScrollView, TouchableOpacity, Dimensions, Linking } from 'react-native';
import { WebView } from 'react-native-webview';
import _ from 'lodash';
import email from 'react-native-email';
import Icon from 'react-native-vector-icons/FontAwesome';
import IconE from 'react-native-vector-icons/Entypo';

import {playstorelink, githubURL} from "../../external_links";

export class Feedback extends Component {

  sendMail() {
    email(['joan3pastor@gmail.com'], {
      subject: 'FIB APP - Feedback & Bugs',
    }).catch(() => alert('Error obrint el client de correu... Pots enviar el teus comentaris a joan.marc.pastor@est.fib.upc.edu'));
  }

  redirectToPlayStore() {
    Linking.openURL(playstorelink);
  }

  render() {
    var { width, height } = Dimensions.get('window');
    return (
      <View style={{flex:1, alignItems:'center', backgroundColor:'#fefefe', padding:10}}>
          <View style={{width:width-20, height:height/2-90}}>
            <WebView  textZoom={100} 
            scalesPageToFit={false}
            source={{
              html: this.props.lang.text
            }}/>
          </View>

          <TouchableOpacity onPress={this.redirectToPlayStore} activeOpacity={0.8}>
              <View style={{marginTop:30, width:220, flexDirection:'row', alignItems:'center', justifyContent:'space-between', backgroundColor:'#00671D', borderRadius:5, padding:8}}>
                <IconE name="google-play" color='#fff' size={25} style={{marginLeft:5}}/>
                <View style={{flex:1}}>
                  <Text style={{marginRight:0, fontFamily: 'Arial', fontSize: 16, fontWeight:'bold', color:'#fff', textAlign:"center"}}>{`  ${this.props.lang.bttnValora}  ‚ù§   `}</Text>
                </View>
              </View>
            </TouchableOpacity>

            <TouchableOpacity onPress={this.sendMail} activeOpacity={0.8}>
              <View style={{marginTop:30, width:220, flexDirection:'row', alignItems:'center', justifyContent:'space-between', backgroundColor:'#B40404', borderRadius:5, padding:8}}>
                <Icon name="envelope-o" color='#fff' size={25} style={{marginLeft:5}}/>
                <View style={{flex:1}}>
                  <Text style={{marginRight:0, fontFamily: 'Arial', fontSize: 16, fontWeight:'bold', color:'#fff', textAlign:"center"}}>{`  ${this.props.lang.bttnSendMail}  `}</Text>
                </View>
              </View>
            </TouchableOpacity>

            <TouchableOpacity onPress={() => Linking.openURL(githubURL)} activeOpacity={0.8}>
              <View style={{marginTop:30, width:220, flexDirection:'row', alignItems:'center', justifyContent:'space-between', backgroundColor:'#064283', borderRadius:5, padding:8}}>
                <Icon name="github" color='#fff' size={25} style={{marginLeft:5}}/>
                <View style={{flex:1}}>
                  <Text style={{marginRight:0, fontFamily: 'Arial', fontSize: 16, fontWeight:'bold', color:'#fff', textAlign:"center"}}>{`  ${this.props.lang.bttnGithub}  `}</Text>
                </View>
              </View>
            </TouchableOpacity>

          <Text style={{position:'absolute', bottom:20}}>{`  ${this.props.lang.footer}  `}</Text>
        </View>
    );
  }
}


function mapStateToProps(state) {
  return {
    lang: state.lang.feedback,
  };
}

export default connect(mapStateToProps, {  })(Feedback);
