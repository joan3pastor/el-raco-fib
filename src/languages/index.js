import {ca} from "./ca.js";
import {es} from "./es.js";
import {en} from "./en.js";

export default {
    ca: ca,
    es: es,
    en: en,
};