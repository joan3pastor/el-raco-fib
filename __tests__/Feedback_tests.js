import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import axios from 'axios'

// LANG
import {ca} from "../src/languages/ca";
const lang = ca.feedback;

import {Feedback} from "../src/components/navigation/Feedback";
import {playstorelink, githubURL} from "../src/external_links";

Enzyme.configure({ adapter: new Adapter() });
const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {};

beforeEach( () => {})

describe('Testing Feedback component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <Feedback store={mockStore(initialState)} lang={lang} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it("playstorelink url returns 200[OK] code", async () => {
    resp = await axios.get(playstorelink);
    expect(resp.status == 200).toBe(true);
  });
  it("github url returns 200[OK] code", async () => {
    resp = await axios.get(githubURL);
    expect(resp.status == 200).toBe(true);
  });
});

