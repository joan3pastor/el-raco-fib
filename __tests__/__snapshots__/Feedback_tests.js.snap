// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Feedback component renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fefefe",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "height": 577,
        "width": 730,
      }
    }
  >
    <WebView
      cacheEnabled={true}
      originWhitelist={
        Array [
          "http://*",
          "https://*",
        ]
      }
      scalesPageToFit={false}
      source={
        Object {
          "html": "<style>p{text-align:justify}</style><p>Siéntete libre de enviarme un correo si algo no se comporta como esperabas o si echas en falta alguna funcionalidad. Adjunta capturas de pantalla de todo lo posible.</p><p>Trataré de solucionar errores críticos. Todo lo demás dependerá de mi disponibilidad.</p>",
        }
      }
      textZoom={100}
      useSharedProcessPool={true}
      useWebKit={true}
    />
  </View>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00671D",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 30,
          "padding": 8,
          "width": 220,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#fff"
        name="google-play"
        size={25}
        style={
          Object {
            "marginLeft": 5,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 0,
              "textAlign": "center",
            }
          }
        >
            Valora la App  ❤   
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#B40404",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 30,
          "padding": 8,
          "width": 220,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#fff"
        name="envelope-o"
        size={25}
        style={
          Object {
            "marginLeft": 5,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 0,
              "textAlign": "center",
            }
          }
        >
            Envíame un correo!  
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#064283",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 30,
          "padding": 8,
          "width": 220,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#fff"
        name="github"
        size={25}
        style={
          Object {
            "marginLeft": 5,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontFamily": "Arial",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 0,
              "textAlign": "center",
            }
          }
        >
            Colabora!  
        </Text>
      </View>
    </View>
  </TouchableOpacity>
  <Text
    style={
      Object {
        "bottom": 20,
        "position": "absolute",
      }
    }
  >
      Creada por Joan Marc Pastor con la API del Racó  
  </Text>
</View>
`;
